### PR TITLE
#8334 Review mapType configuration and docs

### PR DESCRIPTION
**Pull request by allyoucanmap**
## Description

The PR #8990 introduced the new configuration of mapType but it does not take into account the default visualization mode that for some custom downstream project could be needed (eg initialize the app only in 3D)

New proposed configuration for mapType is:
```js
{
  // allow to define the default visualization mode of the app and
  // which 2D or 3D map library should be used based on the device
  // the configuration below is the default one
  // note: this configuration does not support expressions
  "mapType": {
    // the default visualization mode of the app, it could be "2D" or "3D"
    "defaultVisualizationMode": "2D",
    // map library to use based on the visualization mode and device
    // structure -> { visualizationModes: { [visualizationMode]: { [deviceType]: mapLibrary } } }
    "visualizationModes": {
      "2D": {
        "desktop": "openlayers",
        "mobile": "leaflet"
      },
      "3D": {
        "desktop": "cesium",
        "mobile": "cesium"
      }
    }
  },
}
```

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Minor changes to existing features

## Issue

**What is the current behavior?**

#8334

**What is the new behavior?**

Review of new mapType configuration

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [x] Yes, and I documented them in migration notes

### CODE REVIEW

**Review by offtherailz — CHANGES_REQUESTED** [review]
See only my comment about migration guidelines.

File: docs/developer-guide/mapstore-migration-guide.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ This is a list of things to check if you want to update from a previous version
 The map configuration stores the information related to the visualization mode 2D or 3D after saving a map.
 This update include also following changes:
 
-- `maptype` configuration inside the initialState of localConfig needs to be removed in favor of the global mapType configuration
+- `maptype` configuration inside the initialState of localConfig needs to be removed in favor of the global mapType configuration. The global mapType configuration is only needed for project with custom map library settings.
```
Comment (offtherailz) [inline docs/developer-guide/mapstore-migration-guide.md:30]: Please remove the sectuon below and add a URL here
```suggestion
- `maptype` configuration inside the initialState of `localConfig.json` needs to be removed. If you were using this configuration to customize the map library, refer to [this documentation](url) to update your configuration.
```

File: docs/developer-guide/mapstore-migration-guide.md [hunk 2]
```diff
@@ -44,30 +44,35 @@ This update include also following changes:
...
++       "visualizationModes": {
++           "2D": {
++               "desktop": "openlayers",
++               "mobile": "openlayers"
++           },
++           "3D": {
++               "desktop": "cesium",
++               "mobile": "cesium"
++           }
 +       }
```
Comment (offtherailz) [inline docs/developer-guide/mapstore-migration-guide.md:66]:
```suggestion
```

File: docs/developer-guide/mapstore-migration-guide.md [hunk 3]
```diff
@@ -27,7 +27,7 @@ This is a list of things to check if you want to update from a previous version
 The map configuration stores the information related to the visualization mode 2D or 3D after saving a map.
 This update include also following changes:
 
-- `maptype` configuration inside the initialState of localConfig needs to be removed in favor of the global mapType configuration
+- `maptype` default state configuration inside the initialState of localConfig.json needs to be removed. If a MapStore project needs a particular setup (eg. use only OpenLayers for 2D maps, initialize the app in 3D, ...) it is possible to override the default map libraries configuration with the new `mapType` property in the localConfig.json file, see documentation [here](local-config.md#application-configuration).
```
Comment (offtherailz) [inline docs/developer-guide/mapstore-migration-guide.md:30]:
```suggestion
- `maptype` default state configuration inside the initialState of localConfig.json needs to be removed. If a MapStore project needs a particular setup (eg. use only OpenLayers for 2D maps, initialize the app in 3D, ...) it is possible to override the default map libraries configuration with the new `mapType` property in the `localConfig.json` file, see documentation [here](local-config.md#application-configuration).
```

Comment (offtherailz) [inline docs/developer-guide/mapstore-migration-guide.md:30]:
```suggestion
- `maptype` default state configuration inside the initialState of `localConfig.json` needs to be removed. If a MapStore project needs a particular setup (eg. use only OpenLayers for 2D maps, initialize the app in 3D, ...) it is possible to override the default map libraries configuration with the new `mapType` property in the localConfig.json file, see documentation [here](local-config.md#application-configuration).
```

Comment (offtherailz) [inline docs/developer-guide/mapstore-migration-guide.md:30]:
```suggestion
- `maptype` default state configuration inside the initialState of `localConfig.json` needs to be removed. If a MapStore project needs a particular setup (eg. use only OpenLayers for 2D maps, initialize the app in 3D, ...) it is possible to override the default map libraries configuration with the new `mapType` property in the `localConfig.json` file, see documentation [here](local-config.md#application-configuration).
```

**Review by offtherailz — APPROVED** [review]

**Comment by allyoucanmap** [comment]
@ElenaGallo please test this fix on dev as part of the work done in the PR #8990, thanks

**Pull request merged**